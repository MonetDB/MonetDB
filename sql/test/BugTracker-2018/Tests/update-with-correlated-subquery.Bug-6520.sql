start transaction;

CREATE TABLE "PRIMARY" (
  "ID" integer NOT NULL,
  "LEN" real NOT NULL,
  CONSTRAINT "PK_PRIMARY" PRIMARY KEY ("ID")
);

CREATE TABLE "FOREIGN" (
  "ID" integer NOT NULL,
  "POS" real NOT NULL,
  CONSTRAINT "FK_FOREIGN" FOREIGN KEY ("ID") REFERENCES "PRIMARY" ("ID")
);

alter table "FOREIGN" add column "RelPos" real default NULL;

update "FOREIGN"
	set "RelPos" = "POS" / (
		select "LEN"
		from "PRIMARY"
		where "PRIMARY"."ID" = "FOREIGN"."ID"
	)
	where "POS" between 0 and (
		select "LEN"
		from "PRIMARY"
		where "PRIMARY"."ID" = "FOREIGN"."ID"
	)
;

rollback;
