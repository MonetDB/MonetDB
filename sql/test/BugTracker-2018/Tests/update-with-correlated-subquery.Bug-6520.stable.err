stderr of test 'update-with-correlated-subquery.Bug-6520` in directory 'sql/test/BugTracker-2018` itself:


# 18:27:51 >  
# 18:27:51 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=32298" "--set" "mapi_usock=/var/tmp/mtest-6807/.s.monetdb.32298" "--set" "monet_prompt=" "--forcemito" "--dbpath=/ufs/manegold/_/Monet/preview/prefix/_/var/MonetDB/mTests_sql_test_BugTracker-2018"
# 18:27:51 >  

# builtin opt 	gdk_dbpath = /ufs/manegold/_/Monet/preview/prefix/_/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 32298
# cmdline opt 	mapi_usock = /var/tmp/mtest-6807/.s.monetdb.32298
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /ufs/manegold/_/Monet/preview/prefix/_/var/MonetDB/mTests_sql_test_BugTracker-2018
# cmdline opt 	gdk_debug = 553648138

# 18:27:52 >  
# 18:27:52 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-6807" "--port=32298"
# 18:27:52 >  


# 18:27:52 >  
# 18:27:52 >  "Done."
# 18:27:52 >  

