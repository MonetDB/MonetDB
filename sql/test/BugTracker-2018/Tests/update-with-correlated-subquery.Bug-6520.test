statement ok
start transaction

statement ok
CREATE TABLE "PRIMARY" (
  "ID" integer NOT NULL,
  "LEN" real NOT NULL,
  CONSTRAINT "PK_PRIMARY" PRIMARY KEY ("ID")
)

statement ok
CREATE TABLE "FOREIGN" (
  "ID" integer NOT NULL,
  "POS" real NOT NULL,
  CONSTRAINT "FK_FOREIGN" FOREIGN KEY ("ID") REFERENCES "PRIMARY" ("ID")
)

statement ok
alter table "FOREIGN" add column "RelPos" real default NULL

statement ok
update "FOREIGN"
	set "RelPos" = "POS" / (
		select "LEN"
		from "PRIMARY"
		where "PRIMARY"."ID" = "FOREIGN"."ID"
	)
	where "POS" between 0 and (
		select "LEN"
		from "PRIMARY"
		where "PRIMARY"."ID" = "FOREIGN"."ID"
	)

statement ok
rollback

