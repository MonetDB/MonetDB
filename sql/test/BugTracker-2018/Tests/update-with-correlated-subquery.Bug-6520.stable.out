stdout of test 'update-with-correlated-subquery.Bug-6520` in directory 'sql/test/BugTracker-2018` itself:


# 18:27:51 >  
# 18:27:51 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=32298" "--set" "mapi_usock=/var/tmp/mtest-6807/.s.monetdb.32298" "--set" "monet_prompt=" "--forcemito" "--dbpath=/ufs/manegold/_/Monet/preview/prefix/_/var/MonetDB/mTests_sql_test_BugTracker-2018"
# 18:27:51 >  

# MonetDB 5 server v11.27.12
# This is an unreleased version
# Serving database 'mTests_sql_test_BugTracker-2018', using 4 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 31.309 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2018 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://earth.da.cwi.nl:32298/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-6807/.s.monetdb.32298
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded


# 18:27:52 >  
# 18:27:52 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-6807" "--port=32298"
# 18:27:52 >  

#start transaction;
#CREATE TABLE "PRIMARY" (
#  "ID" integer NOT NULL,
#  "LEN" real NOT NULL,
#  CONSTRAINT "PK_PRIMARY" PRIMARY KEY ("ID")
#);
#CREATE TABLE "FOREIGN" (
#  "ID" integer NOT NULL,
#  "POS" real NOT NULL,
#  CONSTRAINT "FK_FOREIGN" FOREIGN KEY ("ID") REFERENCES "PRIMARY" ("ID")
#);
#alter table "FOREIGN" add column "RelPos" real default NULL;
#update "FOREIGN"
#	set "RelPos" = "POS" / (
#		select "LEN"
#		from "PRIMARY"
#		where "PRIMARY"."ID" = "FOREIGN"."ID"
#	)
#	where "POS" between 0 and (
#		select "LEN"
#		from "PRIMARY"
#		where "PRIMARY"."ID" = "FOREIGN"."ID"
#	)
#;
[ 0	]
#rollback;

# 18:27:52 >  
# 18:27:52 >  "Done."
# 18:27:52 >  

